package.json

```diff
@@ -2,104 +2,98 @@
   "name": "nativefier",
   "version": "7.7.1",
   "description": "Wrap web apps natively",
+  "license": "MIT",
+  "author": "Goh Jia Hao",
+  "engines": {
+    "node": ">= 8.10.0",
+    "npm": ">= 5.6.0"
+  },
   "keywords": [
     "desktop",
     "electron",
     "app",
     "native",
     "wrapper"
   ],
-  "main": "lib/index.js",
-  "scripts": {
-    "dev-up": "npm install && (cd ./app && npm install) && npm run build",
-    "dev-up-win": "npm install & cd app & npm install & cd .. & npm run build",
-    "test": "jest src",
-    "guard": "jest --watch src",
-    "e2e": "jest e2e",
-    "tdd": "gulp tdd",
-    "lint": "eslint .",
-    "lint:fix": "eslint . --fix",
-    "ci": "npm run lint && npm test && npm run e2e",
-    "clean": "gulp clean",
-    "build": "gulp build",
-    "watch": "while true ; do gulp watch ; done",
-    "package-placeholder": "npm run build && node lib/cli.js http://www.bennish.net/web-notifications.html ~/Desktop --overwrite --name notification-test --icon ./test-resources/iconSampleGrey.png --inject ./test-resources/test-injection.js --inject ./test-resources/test-injection.css && open ~/Desktop/notification-test-darwin-x64/notification-test.app",
-    "start-placeholder": "npm run build && electron app",
-    "changelog": "./scripts/changelog",
-    "format": "prettier --write '{gulp,src}/**/*.js' 'app/src/**/*.js'"
-  },
+  "main": "lib/src/main.js",
   "bin": {
-    "nativefier": "lib/cli.js"
+    "nativefier": "lib/src/cli.js"
   },
+  "homepage": "https://github.com/jiahaog/nativefier",
   "repository": {
     "type": "git",
     "url": "git+https://github.com/jiahaog/nativefier.git"
   },
-  "author": "Goh Jia Hao",
-  "license": "MIT",
   "bugs": {
     "url": "https://github.com/jiahaog/nativefier/issues"
   },
-  "homepage": "https://github.com/jiahaog/nativefier#readme",
+  "scripts": {
+    "build-app-static": "ncp app/src/static/ app/lib/static/",
+    "build-app": "cd app && tsc",
+    "build-cli": "tsc",
+    "build": "npm run clean && npm run build-cli && npm run build-app && npm run build-app-static",
+    "changelog": "./docs/generate-changelog",
+    "ci": "npm run lint && npm test && npm run test:integration",
+    "clean": "rimraf lib/ app/lib/",
+    "clean:full": "rimraf lib/ app/lib/ node_modules/ app/node_modules/",
+    "compile:watch": "tsc -w --noUnusedLocals false",
+    "compile:watch:app": "cd app && tsc -w --noUnusedLocals false",
+    "lint:fix": "eslint . --fix",
+    "lint:format": "prettier --write 'src/**/*.js' 'app/src/**/*.js'",
+    "lint": "eslint . --ext .ts",
+    "list-outdated-deps": "npm out; cd app && npm out; true",
+    "postinstall": "cd app && npx yarn install --no-lockfile --prefer-offline --prod --no-progress --silent",
```

Comment:
> I'm wondering if using any of the other [hooks](https://docs.npmjs.com/misc/scripts) besides `postinstall` will allow us to workaround the issue.

@jiahaog I tried `preinstall`, it exhibited the same issue.

> I'd prefer to including `yarn` as a dependency below and calling it directly instead of using `npx` to do so, so that we at least have some control over the versioning of the tool

@jiahaog true. Doing so.